### PR TITLE
Multithreading cleanup

### PR DESCRIPTION
**Pull request by devgrok**
This is a complete code reveamp.
Due to what appeared to be multithreading issues - if you have 2 different clients access different channels then they conflict with each other. To rule that out, I refactored the AudioAddict class, changing it into a more object-orientated design (instead of using it as just a helper class/namespace)

### CODE REVIEW

**Comment by phrawzty** [comment]
Hello! First off, thanks for the big PR - to be honest, I sort of thought that I was the only person that used this. 😄  You've made a lot of modifications, so it'll take me a little while to review, but I'm stoked that you went through all the effort!

Question: Does this change any functionality or add/remove features, or is it a straight re-factoring with multi-threading in mind?

File: Contents/Code/audioaddict.py [hunk 1]
```diff
@@ -2,97 +2,111 @@
...
+if "Plugin" not in globals():
+    from plexpluginstub.stubs import *
+    sys.stderr.write("Imported stub code and shouldn't have\n")
+#################################################
+
+
+# Valid streaming services according to audioaddict.com.
+validServices = {
+    'radiotunes': {'name':'RadioTunes.com', 'listen':'listen.radiotunes.com', 'image':'http://cdn.audioaddict.com/radiotunes.com/assets/android_app_banner/icon-c9f306a785072b952e1332f652164148.png'},
+    'di':         {'name':'DI.fm', 'listen':'listen.di.fm', 'image':'http://cdn.audioaddict.com/di.fm/assets/android_app_banner/icon-36f5a36bcd7f7ee5c58bdc608a006d19.png'},
```
Comment (phrawzty) [inline Contents/Code/audioaddict.py]: Oddly, in Plex for Samsung, this image (and not the others) isn't constrained within the display box. Weird!

**Comment by phrawzty** [comment]
I haven't really started looking at the code; however, I can say this right away: this branch breaks functionality on my Samsung TV (Plex for Samsung). No idea why quite yet, but that's definitely something I'll have to investigate. If you've got any ideas or insight, I'm all ears. 😄

**Comment by devgrok** [comment]
I forgot to reply earlier - it appears I forgot to commit a change. 
I also have a Samsung (with the Plex for Samsung app) which is what I use it on :)

**Comment by phrawzty** [comment]
I haven't had much free time at all to review this (sorry!), but if you've got additional commits that fix Plex for Samsung functionality, *please update this PR*. I'm all about improving the user experience in this plugin. 😄

**Pull request closed without merging**